### ISSUE TITLE
`no-unused-colocally-location` rule

### ISSUE THREAD

**Issue by shumbo** [shumbo/choreography-ts#7]
Write a rule that warns the user if they put more locations than necessary to `colocally`.

```ts
type Locations = "alice" | "bob" | "carol"

const test: Choreography<Locations> = async ({
  locally,
  broadcast,
  colocally,
}) => {
  const msgAtCarol = await locally("carol", () => "I'm Carol");
  await colocally(["alice", "bob"], async () => {
    await locally("alice", () => {
      console.log("hello from alice")
    })
    return [];
  });
  return [];
};
```

In the above example, `colocally` is executed on Alice and Bob, but only Alice is used inside the body.

**Comment by ImpregnableProgrammer** [comment]
This is being worked on in pull request #29

**Comment by shumbo** [comment]
Resolved with #29
